packages/devtools_app/lib/src/debugger/codeview.dart

```diff
@@ -298,15 +312,39 @@ class _CodeViewState extends State<CodeView>
                         Expanded(
                           child: LayoutBuilder(
                             builder: (context, constraints) {
-                              return Lines(
-                                constraints: constraints,
-                                scrollController: textController,
-                                lines: lines,
-                                pausedFrame: pausedFrame,
-                                searchMatchesNotifier:
-                                    widget.controller.searchMatches,
-                                activeSearchMatchNotifier:
-                                    widget.controller.activeSearchMatch,
+                              final num longestLineLength = lines.fold(
+                                  0,
+                                  (longestLength, line) => math.max(
+                                      longestLength,
+                                      line.toPlainText().length));
+                              final fileWidth =
+                                  CodeView.assumedCodeCharacterWidth *
+                                      longestLineLength;
+                              final boxWidth = math.max(
+                                  constraints.minWidth, constraints.maxWidth);
+
+                              return Scrollbar(
+                                isAlwaysShown: fileWidth > boxWidth,
```

Comment:
What does it look like if you set this to true on a narrow file?

Reply:
It shows up at the bottom even though the file isn't scrollable:
![image](https://user-images.githubusercontent.com/21270878/129094457-408f701e-3d07-4f4d-a85c-548c5c3e8bcb.png)

This also happens on hover currently unfortunately (scrollbar shows up when hovering at the bottom of the file even if it's not scrollable). I'm looking into why that is - once I fix that I think it would make sense to set this to `true`.

Reply:
the scrollbar seems like it is the wrong size. why isn't the scrollbar the entire width of the window given the content is no larger than the size of the window? what happens if you scroll in that case?

Reply:
Ah I see what the issue is. This happens when I open a file with the `Libraries` button, which opens the side panel. With the side panel open, the horizontal scrollbar is necessary. When I then close the side panel, the scrollbar isn't redrawn.

Reply:
hmm. Is the LayoutBuilder called again when the Libraries panel is closed? I would think it is.

Reply:
Ah interesting, it is being called. But the scrollbar isn't being removed, even with the new constraints. This looks similar to some issues that were marked as being recently fixed with https://github.com/flutter/flutter/pull/82687. I'm currently using the stable/official Flutter SDK, which doesn't have that change yet. I'm going to switch to a local fork of `flutter` and see if this issue goes away.

Reply:
Resolving this thread since https://github.com/flutter/flutter/pull/88152 should fix it